### PR TITLE
Issue #1207

### PR DESCRIPTION
**Pull request by AbfrmBlr**
Putting ehcache state specifc service in ehcache project only, Need to decide whether this can be moved/even generalised and moved to terracotta-platform.

### CODE REVIEW

**Comment by albinsuresh** [comment]
Most of the formatting comments given in https://github.com/Terracotta-OSS/terracotta-platform/issues/121 applies here as well.

**Comment by lorban** [comment]
Why all the code commenting? If some code isn't needed anymore, remove it, don't comment it out.

**Comment by AbfrmBlr** [comment]
Oh for sure, just ignore for it now. I have just kept them to make sure the tests line I have commented are covered in EhcacheActiveEntityTest in the next commit, then I will squash that commit.

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
 import org.ehcache.clustered.lock.server.VoltronReadWriteLockServerEntityService;
 import org.ehcache.clustered.server.EhcacheServerEntityService;
 
+import org.ehcache.clustered.server.state.EntityStateServiceProvider;
```
Comment (ljacomet) [inline clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java]: That can be reverted

**Comment by AbfrmBlr** [comment]
All the formatting comments are taken care of

File: clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java [hunk 2]
```diff
@@ -56,36 +57,49 @@
...
+  private static final String SHARED_RESOURCE_POOLS = "sharedResourcePools";
+  private static final String DEDICATED_RESOURCE_POOLS = "dedicatedResourcePools";
+  private static final String STORES = "stores";
+  private static final String IS_CONFIGURED = "true";
+
   private final ServiceRegistry services;
   private final Set<String> offHeapResourceIdentifiers;
 
   /**
    * The name of the resource to use for dedicated resource pools not identifying a resource from which
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java]: This javadoc no longer matches the field

File: clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java [hunk 3]
```diff
@@ -156,16 +170,18 @@ void configure(ConfigureStoreManager message) throws ClusterException {
...
+        this.defaultServerResource.put(DEFAULT_SERVER_RESOURCE, defaultServerResource);
+        if (!offHeapResourceIdentifiers.contains(defaultServerResource)) {
+          throw new ResourceConfigurationException("Default server resource '" + defaultServerResource
                                                    + "' is not defined. Available resources are: " + offHeapResourceIdentifiers);
         }
       }
 
-      this.sharedResourcePools = createPools(resolveResourcePools(configuration));
-      this.stores = new HashMap<String, ServerStoreImpl>();
+      this.defaultServerResource.put(IS_CONFIGURED, IS_CONFIGURED);
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java]: Pretty sure that's a bug right there

File: clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java [hunk 4]
```diff
@@ -56,36 +57,49 @@
...
+
   private final ServiceRegistry services;
   private final Set<String> offHeapResourceIdentifiers;
 
   /**
    * The name of the resource to use for dedicated resource pools not identifying a resource from which
    * space for the pool is obtained.  This value may be {@code null};
    */
-  private String defaultServerResource;
+  private final Map<String, String> defaultServerResource;
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreManager.java]: A rename of the field will help identify what it has become

**Comment by ljacomet** [comment]
Had a chat with @AbfrmBlr and @albinsuresh where we discussed simply making the ServerStoreManager the value returned by a dedicated Ehcache service to manage state.

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 5]
```diff
@@ -264,6 +226,10 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
                                                   " Check your server configuration and define at least one offheap resource.");
       }
 
+      if (this.ehcacheStateService == null) {
+        throw new ServerMisconfigurationException("Server failed to retrieve EhcacheStateService .");
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Let's throw an `AssertionError` in the constructor for this - if it happens, it is serious.

**Pull request merged**